src/common/Surface.js

```diff
@@ -216,7 +221,35 @@
             ;
         }
 
-        this._menu.element().node().parentNode.appendChild(this._menu.element().node());
+        //this._menu.element().node().parentNode.appendChild(this._menu.element().node()); WHAT DOES THIS DO?
+
+        var buttonClientSize = this.showIcon() ? this._icon.getBBox(true) : {width:0, height: 0};
+
+        this._buttonArr = [];
+        this.buttonAnnotations().forEach(function(button, i) {
+            context._buttonArr.push(new Icon().faChar(button['char']).paddingPercent(50))
+        });
```

Comment:
i tried this and got it working, however creating the widgets inside d3 cycle messes up the var/context that gets passes into the onclick callback function, ill will do a seperate PR showing you what i mean